eventsources/sources/generic/start.go

```diff
@@ -0,0 +1,113 @@
+package generic
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/argoproj/argo-events/common/logging"
+	"github.com/argoproj/argo-events/eventsources/sources"
+	apicommon "github.com/argoproj/argo-events/pkg/apis/common"
+	"github.com/argoproj/argo-events/pkg/apis/events"
+	"github.com/argoproj/argo-events/pkg/apis/eventsource/v1alpha1"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+)
+
+// EventListener implements Eventing for generic event source
+type EventListener struct {
+	EventSourceName    string
+	EventName          string
+	GenericEventSource v1alpha1.GenericEventSource
+	conn               *grpc.ClientConn
+}
+
+// GetEventSourceName returns name of event source
+func (el *EventListener) GetEventSourceName() string {
+	return el.EventSourceName
+}
+
+// GetEventName returns name of event
+func (el *EventListener) GetEventName() string {
+	return el.EventName
+}
+
+// GetEventSourceType return type of event server
+func (el *EventListener) GetEventSourceType() apicommon.EventSourceType {
+	return apicommon.GenericEvent
+}
+
+// StartListening listens to generic events
+func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byte) error) error {
+	logger := logging.FromContext(ctx).
+		With(zap.String(logging.LabelEventSourceType, string(el.GetEventSourceType())),
+			zap.String(logging.LabelEventName, el.GetEventName()),
+			zap.String("url", el.GenericEventSource.URL))
+	logger.Info("started processing the generic event source...")
+	defer sources.Recover(el.GetEventName())
+
+	logger.Info("connecting to eventsource server in 5 seconds...")
+	ticker := time.NewTicker(5 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			logger.Info("closing client connection and exiting eventsource...")
+			el.conn.Close()
```

Comment:
Minor - need to check if `el.conn` is not nil and connected.

Reply:
good catch.